### ISSUE TITLE
Feature request: "reset" for tempo-based, oscillating effects

### ISSUE THREAD

**Issue by cyrusadkisson** [brummer10/guitarix#25]
(Not a bug, but a feature request.)

I use guitarix almost exclusively inside Ardour, relying heavily on automation points to toggle and adjust the effects as the song plays. I'm having an issue when I need an effect that has its own internal tempo (chorus, flange, tremolo and vibe) matched to the tempo of the song.

But I'm NOT asking for tempo matching (although that would be nice). My issue is something else. Read on.

What's happening is there will be a part of the song where, say, tremolo is enabled, but the trem is not at the correct point in its sine/triangle curve. Even if I get the trem effect perfectly matched to the tempo of the song, the "enable" signal may catch the trem in the trough of its sine wave.

Even if the trem tempo were 100% perfectly matched to the tempo of the song, the user can start playback at any time. So it's impossible to know at which part of the sine wave the tremolo effect will be when it comes time for it to be enabled.

What I need is a "reset" switch on these types of effects so that I can "reset" the sine wave (or triangle or whatever) at the same time the effect is toggled on.

I hope I explained this well enough. Image attached. https://imgur.com/3MVIqrb Let me know if I need to try again.

**Comment by brummer10** [comment]
So, you need a switch which reset the osc wave to the start point, as if it just starts when ever you reset. Is that right?
I could add that true. Just let me know if I understand you correct on this.

regards
hermann

**Comment by cyrusadkisson** [comment]
Yes, that's correct. Applies to anything with a sine wave or oscillation like chorus, flange, leslie, trem, vibe, etc.

Delay has a tempo, but it's fine since it always starts from the generation of the sound.

Thanks!

**Comment by brummer10** [comment]
Okay, here is the one for GxTremolo.lv2
Please checkout and test it, if it is what you need I could add it to the others you need (please attach a list)

regards
hermann

**Comment by magnetophon** [comment]
I also think this is a very useful feature!

Unfortunately, when I hit the reset button in GxTremolo.lv2, the plugin breaks: there is silence with maximum depth, and something like ring-modulation at minimum depth.
